applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -903,64 +1021,13 @@ static struct bt_conn_cb conn_callbacks = {
  * @brief	Callback handler for play/pause.
  *
  * @note	This callback is called from MCS after receiving a
- *		command from the client or the local media player.
+ *			command from the client or the local media player.
  *
  * @param[in]	play  Boolean to indicate if the stream should be resumed or paused.
  */
 static void le_audio_play_pause_cb(bool play)
 {
-	int ret;
-
-	LOG_DBG("Play/pause cb, state: %d", play);
-
-	if (play) {
-		if (audio_streams[AUDIO_CH_L].ep->status.state ==
-		    BT_AUDIO_EP_STATE_QOS_CONFIGURED) {
-			ret = bt_audio_stream_enable(&audio_streams[AUDIO_CH_L],
-						     lc3_preset_nrf5340.codec.meta,
-						     lc3_preset_nrf5340.codec.meta_count);
-
-			if (ret) {
-				LOG_WRN("Failed to enable left stream");
-			}
-		}
-
-#if !CONFIG_STREAM_BIDIRECTIONAL
-		if (audio_streams[AUDIO_CH_R].ep->status.state ==
-		    BT_AUDIO_EP_STATE_QOS_CONFIGURED) {
-			ret = bt_audio_stream_enable(&audio_streams[AUDIO_CH_R],
-						     lc3_preset_nrf5340.codec.meta,
-						     lc3_preset_nrf5340.codec.meta_count);
-
-			if (ret) {
-				LOG_WRN("Failed to enable right stream");
-			}
-		}
-#endif /* !CONFIG_STREAM_BIDIRECTIONAL */
-	} else {
-		ret = ctrl_events_le_audio_event_send(LE_AUDIO_EVT_NOT_STREAMING);
-		ERR_CHK(ret);
-
-		if (audio_streams[AUDIO_CH_L].ep->status.state == BT_AUDIO_EP_STATE_STREAMING) {
-			ret = bt_audio_stream_disable(&audio_streams[AUDIO_CH_L]);
-
-			if (ret) {
-				LOG_WRN("Failed to disable left stream");
-			}
-		}
-
-#if !CONFIG_STREAM_BIDIRECTIONAL
-		if (audio_streams[AUDIO_CH_R].ep->status.state == BT_AUDIO_EP_STATE_STREAMING) {
-			ret = bt_audio_stream_disable(&audio_streams[AUDIO_CH_R]);
-
-			if (ret) {
-				LOG_WRN("Failed to disable right stream");
-			}
-		}
-#endif /* !CONFIG_STREAM_BIDIRECTIONAL */
-	}
-
-	playing_state = play;
+	audio_play_pause(play);
```

Comment:
I moved this into a function as if there are additional operations added to the callback then this may break the play/pause required here. Will revert.
Having play pause as the only states is also causing an issue when the system is not yet ready to start a stream and the user presses play. This is because the global variable for the state is set to paused and hence is always in a valid state. If the inactive state was used then this would not happen and we only transition to the play state on reception of data. From then on the button will allow it to toggle between play and pause.

Reply:
Ok, suggest a helper function to check headsets-array to see if we have any ongoing streams:
```C
static bool any_active_headsets_started(void)
{
	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
		if (headsets[i].sink_stream != NULL &&
		    headsets[i].sink_stream->ep->status.state == BT_AUDIO_EP_STATE_STREAMING) {
			return true;
		}
	}

	return false;
}
```
and if there are, then you could pause and play again

Reply:
Yes good idea and I am actually in the process of doing this, similar to the all_active_headsets_started().